nexus/db-queries/src/db/datastore/saga.rs

```diff
@@ -207,16 +200,73 @@ impl DataStore {
 
         Ok(events)
     }
+
+    /// Updates all sagas that are currently assigned to any of the Nexus
+    /// instances in `nexus_zone_ids`, assigning them to `new_sec_id` instead.
+    ///
+    /// This change causes the Nexus instance `new_sec_id` to discover these
+    /// sagas and resume executing them the next time it performs saga recovery
+    /// (which is normally on startup and periodically).  Generally,
+    /// `new_sec_id` is the _current_ Nexus instance and the caller should
+    /// activate the saga recovery background task after calling this function
+    /// to immediately resume the newly-assigned sagas.
+    ///
+    /// **Warning:** This operation is only safe if the Nexus instances
```

Comment:
Is it worth checking that the the nexus zone is expunged here to prevent an issue? On the one hand this seems like a good safety check. On the other hand it seems like we may also end up assigning sagas to another nexus of the same version during rolling upgrade if we don't want to actually wait to drain them.

Reply:
That would be nice but I think it's way beyond the scope of this layer.  The datastore doesn't know anything about blueprints, expungement, etc.  Its caller could, but it's so simple that I don't think an extra check would help:
https://github.com/oxidecomputer/omicron/pull/6215/files#diff-5a1edb7bfa049a98af5ba0e05a8c2455bb637d63d8e59ed738a8e78bd6342ed7R38-R51